spock-specs/src/test/groovy/org/spockframework/smoke/condition/ExceptionConditions.groovy

```diff
@@ -168,6 +168,59 @@ thrown() == e
     e.message.startsWith("Exception conditions are only allowed as top-level statements")
   }
 
+  def "thrown condition with neither inferred nor explicit type"() {
+    when:
+    runner.runFeatureBody """
+when:
+true
+
+then:
+thrown()
+    """
+
+    then:
+    InvalidSpecException e = thrown()
+    e.message == "Thrown exception type cannot be inferred automatically. Please specify a type explicitly (e.g. 'thrown(MyException)')."
+  }
+
+  def "thrown condition with only explicit type null"() {
+    when:
+    runner.runFeatureBody """
+when:
+true
+
+then:
+thrown(null)
+    """
+
+    then:
+    InvalidSpecException e = thrown()
+    e.message == "Thrown exception type cannot be inferred automatically. Please specify a type explicitly (e.g. 'thrown(MyException)')."
```

Comment:
Those 2 cases are similar. Maybe `where:` could be used to reduce duplication (unless, you want to emphasis by the name of the test what case is exactly covered here).